sim-lib/src/lib.rs

```diff
@@ -352,6 +358,16 @@ struct Payment {
     dispatch_time: SystemTime,
 }
 
+impl Payment {
+    fn activity_prefix(&self) -> String {
+        if !self.activity_name.is_empty() {
+            format!("{} activity:", self.activity_name)
```

Comment:
nit: should have a space after `:` otherwise it reads `X activity:Payment` when we use it below